implementations/rust/ockam/ockam_api/src/kafka/integration_test.rs

```diff
@@ -1,14 +1,8 @@
 #[cfg(test)]
 mod test {
-    use crate::hop::Hop;
-    use crate::kafka::protocol_aware::utils::{encode_request, encode_response};
-    use crate::kafka::secure_channel_map::ForwarderCreator;
-    use crate::kafka::{
-        KafkaInletController, KafkaPortalListener, KafkaSecureChannelControllerImpl,
-    };
-    use crate::nodes::registry::KafkaServiceKind;
-    use crate::test::NodeManagerHandle;
-    use crate::DefaultAddress;
+    use std::sync::atomic::{AtomicBool, Ordering};
```

Comment:
Was there a reason to change the order of imports? For example, there is 1 line changed in this, but because of imports there are 39 lines changed.

Reply:
I didn't see that. This is CLion deciding to sort imports alphabetically apparently but I can't find where this setting is coming from

Reply:
I think sorting imports is fine and is enforced by `cargo fmt`. The problem I think that previously imports didn't have any blank lines between them, which resulted in a totally different order

Reply:
I'm not sure about Clion, but these are the settings I know for rustfmt/cargo fmt and rust-analyzer. IME the newlines between logical groupings are significant to both tools, like Sanjo said.

## Rustfmt

These were nightly-only when I started using them but may have stabilized:
https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#group_imports
https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#imports_granularity

```toml
group_imports = "StdExternalCrate" # other options: Preserve, One
imports_granularity = "Crate" # other options: Preserve, Module, Item, One 
```

## Rust Analyzer

https://rust-analyzer.github.io/manual.html#configuration
https://rust-analyzer.github.io/manual.html#auto-import

```json
"rust-analyzer.imports.granularity.group": "crate",
"rust-analyzer.imports.group.enable": true,
"rust-analyzer.imports.prefix": "self"
```
The second link describes other possible values.